src/main/java/uk/gov/hmcts/reform/preapi/tasks/CleanupLiveEvents.java

```diff
@@ -1,39 +1,201 @@
-                        log.info("Stopping live event {}", liveEventDTO.getId());
-                        // @todo uncomment this line when https://github.com/hmcts/pre-api/pull/579/ is merged
-                        // mediaService.stopLiveEvent(liveEventDTO.getId());
-                    });
+        var liveEvents = mediaService.getLiveEvents();
+        liveEvents.stream()
+                  .filter(liveEventDTO -> liveEventDTO
+                      .getResourceState().equals(LiveEventResourceState.RUNNING.toString())
+                  )
+                  .forEach(liveEventDTO -> {
+                      log.info("Finding capture session by live event id {}", liveEventDTO.getName());
+                      var sendNotification = false;
+                      try {
+                          var captureSession = captureSessionService.findByLiveEventId(liveEventDTO.getName());
+                          var search = new SearchRecordings();
+                          log.info("Finding recordings by capture session {}", captureSession.getId());
+                          search.setCaptureSessionId(captureSession.getId());
+                          var recordings = recordingService.findAll(search, false, Pageable.unpaged());
+
+                          if (recordings.isEmpty()) {
+                              log.info("No recordings found for capture session {}", captureSession.getId());
+                              sendNotification = stopLiveEvent(
+                                  mediaService,
+                                  captureSession,
+                                  UUID.randomUUID(),
+                                  liveEventDTO.getName()
+                              );
+                          } else {
+                              sendNotification = !recordings.map(recording -> {
+                                  log.info(
+                                      "{} recordings found for capture session {}",
+                                      recordings.getSize(),
+                                      captureSession.getId()
+                                  );
+                                  return stopLiveEvent(
+                                      mediaService,
+                                      captureSession,
+                                      recording.getId(),
+                                      liveEventDTO.getName()
+                                  );
+                              })
+                                                            .filter(b -> b)
+                                                            .toList()
+                                                            .isEmpty();
+                          }
+
+                          if (sendNotification) {
+                              var booking = bookingService.findById(captureSession.getBookingId());
+                              if (booking != null) {
```

Comment:
`booking` will never be null because the service throws an error if it cannot find the booking